bootstrap/src/hash.mli

```diff
@@ -3,22 +3,41 @@
 open Rudiments_int0
 
 type t
```

Comment:
`Hash.t` remains opaque for the moment because 1) we're not actually reading it for anything yet, and 2) it's going to be a `u128` or the moral equivalent thereof (rather than the current `usize`), and I don't want to expose the innards until the type is right.